R/tools.R

```diff
@@ -26,6 +26,52 @@ lint_dir <- function(path) {
   lints
 }
 
+lint_file <- function(path) {
+  if (interactive()) {
+    message(cli::format_inline("Linting {.file {path}}"))
+  }
+  lints <- lintr::lint(path)
+  # Use the actual path provided by the user (typically relative) to make results more readable.
+  # (`lintr::lint()` normalizes paths, e.g. `app/main.R` becomes a full absolute path.)
+  for (i in seq_along(lints)) {
+    lints[[i]]$filename <- path
+  }
+  lints
+}
+
+lint_path <- function(path) {
+  # Check if path is a directory or a file such that nothing happens when the path is invalid.
+  if (fs::is_dir(path)) {
+    lint_dir(path)
+  } else if (fs::is_file(path)) {
+    lint_file(path)
+  }
+}
+
+check_paths <- function(paths) {
+  paths_exist <- fs::file_access(paths, mode = "exists")
+
+  if (all(!paths_exist)) {
+    cli::cli_abort(
+      c(
+        "Nothing to lint.",
+        i = "Please check that {cli::col_blue('paths')} has at
+          least {cli::style_underline('ONE')} existing file or directory."
+      )
+    )
+  } else if (any(!paths_exist)) {
+    cli::cli_inform(
+      ifelse(
+        length(paths[!paths_exist]) > 1,
+        "The following paths do not exist: {.file {paths[!paths_exist]}}.",
+        "This path does not exist: {.file {paths[!paths_exist]}}."
+      )
```

Comment:
[Pluralization in cli](https://cli.r-lib.org/reference/pluralization.html) is pretty cool:
```suggestion
    cli::cli_inform(
      "The following path{?s} cannot be read: {.file {paths[!readable]}}.",
    )
```